### DISCUSSION TITLE
*.java is a non-project file

### DISCUSSION THREAD

**Discussion by thatvertigo** [mfussenegger/nvim-jdtls#112] [Q&A]
I am trying to configure this plugin and for some reason suddenly with no configuration changes a ton of code information stopped working, along with a message `Lox.java is a non-project file, only syntax errors are reported` appearing at the top of every file. I'm aware this is probably some sort of user error, but how can I make sure that jdtls detects the workspace properly

**Comment by mfussenegger** [comment]
Did you update eclipse.jdt.ls or did maven/gradle get updated within a project? That could have caused the change.
Otherwise you could try following the steps outlined in the [troubleshooting section](https://github.com/mfussenegger/nvim-jdtls#troubleshooting) and verify that you got the [root_dir](https://github.com/mfussenegger/nvim-jdtls#root_dir-configuration) configuration right.

**Reply by thatvertigo** [reply]
My project does not use maven nor gradle, so I initialized a git repository in the project root directory which should be detected, no?

**Reply by IlyasYOY** [reply]
I faced the same issue. Usually I get it fixed cleaning eclipse projects folder & restarting nvim.

**Comment by jeusdi** [comment]
I getting same behavior. I mean, I'm getting same message:

> MainVerticle.java is not on the classpath of project starter, only syntax errors are reported

This is my my `~/.config/nvim/init.vim`:

```
call plug#begin(stdpath('config') . '/plugged')
	Plug 'vim-airline/vim-airline'
	Plug 'neovim/nvim-lspconfig'
	Plug 'glepnir/lspsaga.nvim'
	Plug 'mfussenegger/nvim-jdtls'
	Plug 'kabouzeid/nvim-lspinstall'
call plug#end()

if has('nvim-0.5')
  augroup lsp
    au!
    au FileType java lua require('jdtls').start_or_attach({cmd = {'/home/jeusdi/.local/share/nvim/lspinstall/java/jdtls.sh', '/home/jeusdi/workspace/' .. vim.fn.fnamemodify(vim.fn.getcwd(), ':p:h:t')}})

  augroup end
endif

set relativenumber
```

After `nvim` is started, next folders are created in `/home/jeusdi/workspace`:

```
/home/jeusdi/workspace/vert.x
└── jdt.ls-java-project
    ├── bin
    └── src
```

Any ideas?

**Reply by danielnehrig** [reply]
check solution of #151 i think you have the same problem with your -data path

**Comment by danielnehrig** [comment]
i have the same probem #151 for more info

**Comment by takadev15** [comment]
I had the same problem even tough it's a gradle project. The project itself were a multi-project gradle with multiple build.gradle on each subprojects and settings.gradle on root folder. Weirdly enough this problem only occur on my works project and if i tested in on other open source project with same folder structure it running fine. I already opened an issue for this in #490 for more information regarding this problem
